examples/kubernetes-egress-gateway/egress-nat-policy.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: cilium.io/v2alpha1
+kind: CiliumEgressNATPolicy
+metadata:
+  name: egress-sample
+spec:
+  egress:
+  - podSelector:
+      matchLabels:
+        class: mediabot
+    namespaceSelector:
+      matchLabels:
+        ns: default
```

Comment:
Wouldn't it be simpler to add namespace label match (`io.kubernetes.pod.namespace: default`) to the podSelector instead of using a namespaceSelector? Or is the CiliumEgressNATPolicy somehow applied on the namespace directly, rather than just on the selected pods?

Reply:
namespace label is the field that has direct support from api, which we showed in the guide. `io.kubernetes.pod.namespace: default` probably will also work. @MasterZ40 keep me honest here.

Reply:
So as previously written the watcher is indeed directly invoked by namespace label updates, however I have changed the approach to watch directly on CEP objects for more scalability. Therefore i think @jrajahalme is correct here we should move ns name references to use pod selector labels. I just tested that the "io.kubernetes.pod.namespace: default" approach works for this purpose.

We can also see there the CEP labels with namespace prefix here doesn't actually contain ns name.
```
k8s:class=mediabot
k8s:io.cilium.k8s.namespace.labels.bar=Foo
k8s:io.cilium.k8s.namespace.labels.foo=Bar
k8s:io.cilium.k8s.namespace.labels.user=Alice
k8s:io.cilium.k8s.policy.cluster=default
k8s:io.cilium.k8s.policy.serviceaccount=default
k8s:io.kubernetes.pod.namespace=default
k8s:org=empire
```

Reply:
I fixed this in the newer example.